e2e/smoke/bullseye.lock.json

```diff
@@ -6,7 +6,7 @@
 			"key": "ncurses-base_6.2-p-20201114-2-p-deb11u2_amd64",
 			"name": "ncurses-base",
 			"sha256": "a55a5f94299448279da6a6c2031a9816dc768cd300668ff82ecfc6480bbfc83d",
-			"url": "https://snapshot.debian.org/archive/debian/20240210T223313Z/pool/main/n/ncurses/ncurses-base_6.2+20201114-2+deb11u2_all.deb",
+			"url": "https://snapshot-cloudflare.debian.org/archive/debian/20240210T223313Z/pool/main/n/ncurses/ncurses-base_6.2+20201114-2+deb11u2_all.deb",
```

Comment:
I would expect the lock file to have a `urls` field as well, since urls specified in the manifest is interchangeable and bazel will fallback to next one if one of these links fail. 

So we need a breaking change to the lockfile here.

Reply:
I went with the single URL because I think it's the better source of truth. Meaning, while it's true that the intent of the manifest is to point at equivalent packages, the lock is *the actual stuff downloaded*, and so the `sha256` corresponds to *that exact deb file downloaded* from *that exact URL*. Thus, IMO is actually correct as-is and I think adding multiple URLs wouldn't be right.

WDYT?

Reply:
OK, I'll make the change and will also look into fixing the buildifier errors

Reply:
Its determined at download time.  Lets say you have to urls where an artifact can be fetched, bazel will try both of those urls sequentially until it succeeds, if it can't download it via any of the urls, then it will fail the build. 

But if you determine this a locking time and generate a lockfile with a single location where a package might be downloaded, then Bazel will fail immediately if the said url no longer works.

Reply:
Think of urls array as a list of locations where a package might be fetched, any of them could go offline after locking.

Reply:
Yeah, I mean, I understand and I'm aware of `urls` as a concept and how it's meant to work 😄  I was just considering the lock as being "the actual thing that happened"...  basically, I was considering a "what if" there's a difference in hash between the URLs, but that's also covered, we are using the hash in the lock and passing it to verify the download, in https://github.com/GoogleContainerTools/rules_distroless/blob/2354eb3345e7f332abb0d2ff43bd8399fabca731/apt/extensions.bzl#L39

So all is good 😄 and yeah, having all the possible URLs in the lock is better and more correct.